# enterprise_thruster_node:
#   ros__parameters:
#     # thrusters:
#     vsp_port:
#       type: vsp
#       force_topic: /enterprise/thrusters/port/command
#       arm_x:
#         topic: /enterprise/hardware/servo/servo_1/scaled
#         inverted: true
#       arm_y:
#         topic: /enterprise/hardware/servo/servo_3/scaled
#         inverted: true

#     vsp_starboard:
#       type: vsp
#       force_topic: /enterprise/thrusters/starboard/command
#       arm_x:
#         topic: /enterprise/hardware/servo/servo_4/scaled
#         inverted: false
#       arm_y:
#         topic: /enterprise/hardware/servo/servo_2/scaled
#         inverted: false

#     tunnel_bow:
#       type: fixed
#       force_topic: /enterprise/hardware/servo/bow_thruster/scaled
#       control:
#         inverted: false


vessel_control_node:
  ros__parameters:
    thrusters:

      vsp_port:
        type: vsp
        force_topic: /enterprise/thrusters/port/command
        force_max: 10.0
        force_min: -10.0
        arm_x:
          topic: /enterprise/hardware/servo/servo_1/scaled
          inverted: true
        arm_y:
          topic: /enterprise/hardware/servo/servo_3/scaled
          inverted: true
        rpm:
          topic: /enterprise/hardware/servo/pwm/vsp_1
          cmd: 0.5

      vsp_starboard:
        force_topic: /enterprise/thrusters/starboard/command
        type: vsp
        force_max: 10.0
        force_min: -10.0
        rpm_cmd: 0.5
        arm_x:
          topic: /enterprise/hardware/servo/servo_4/scaled
          inverted: false
        arm_y:
          topic: /enterprise/hardware/servo/servo_2/scaled
          inverted: false
        rpm:
          topic: /enterprise/hardware/servo/pwm/vsp_2
          cmd: 0.5

      tunnel:
        type: fixed
        force_max: 10
        force_min: -10
        force_topic: /enterprise/hardware/servo/bow_thruster/scaled
        control:
          inverted: false